src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Interfaces/Security/UserPublicationRoleRepositoryExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿#nullable enable
+using System;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Common.Services.Security;
+using Microsoft.AspNetCore.Authorization;
+using static GovUk.Education.ExploreEducationStatistics.Content.Model.PublicationRole;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Security
+{
+    public static class UserPublicationRoleRepositoryExtensions
```

Comment:
Not sure what the advantage of using extensions here is? Couldn't we have this directly in the `UserReleaseRoleRepository` ie:

```cs
public Task<bool> IsUserPublicationOwner(Guid userId, Guid publicationId);
```

I'd say using extensions is a bit hide-y and makes the tests more confusing as they hide the methods we're actually calling on the repository. With the current signatures, it also constrains the use of these methods to only the security context, when they could be used in other places too.

Reply:
With the repos having the generic `UserHasRoleOnPublication` , `UserHasRoleOnLatestRelease` methods which have `Guid userId` in their signatures, the extensions were trying to add convenience, avoiding every call needing `User.GetUserId()` from the `AuthorizationContext` and reference the individual role / list of roles. This was while keeping the repository classes small by limiting them to those generic methods. I was also trying to avoid the repos getting bloated with these specific methods where we might eventually have a lot more `IsUser...` permuations, similar to how we have a lot of `CheckCan...` methods on `UserService`.

We see the same approach being done in `Admin.Services.Interfaces.Security.UserServiceExtensions`.

I see your point about the extension classes hiding away the logic so I have removed them and moved the methods into the repository classes now.

The test  setup code now matches the methods which can be seen getting called in the auth handler code 🙂 .

Reply:
Yeh I guess we should try and avoid adding too many extra methods to `UserService` for every permutation of roles. I'd say it's probably not likely we'll need many more in the future though. 

Think it at least makes sense to add extra methods if they're being used over and over again (like `IsUserApproverOnLatestRelease` is)

Reply:
Thanks, I agree we could transition some of them away from extensions into the actual service.